### ISSUE TITLE
bug: explicit-preview-rules enables rule from ALL

### ISSUE THREAD

**Issue by nstarman** [astral-sh/ruff#9536]
In https://github.com/astropy/astropy/pull/15889 I tried turning on `preview` mode with `explicit-preview-rules`. From https://docs.astral.sh/ruff/preview/#selecting-single-preview-rules I would expect that running `ruff lint` (really `pre-commit run ruff -a`) shouldn't do anything since no rules are explicitly selected (we use `select = ["ALL"]`). However ruff finds and fixes `SIM910`.
Is this a bug?

**Comment by charliermarsh** [comment]
It does sound like a bug.

**Comment by charliermarsh** [comment]
Ahh I see... That rule _isn't_ in preview, but we changed the behavior of the rule such that in preview, it flags cases like:

```python
ages = {"Tom": 23, "Maria": 23, "Dog": 11}
age = ages.get("Cat", None)
```

Whereas in stable, it will only flag `.get` calls on literals, like:

```python
{"Tom": 23, "Maria": 23, "Dog": 11}.get("Cat", None)
```

I think this behavior is correct (you're enabling the rule, and `--preview` is modifying its behavior) albeit a bit unclear.

**Comment by charliermarsh** [comment]
\cc @zanieb for visibility, if you're interested

**Comment by zanieb** [comment]
We should probably document preview behaviors better per rule, but I think this is fair expected behavior.

**Comment by charliermarsh** [comment]
Yeah. It actually _is_ mentioned in the [rule documentation](https://docs.astral.sh/ruff/rules/dict-get-with-none-default/), but perhaps it should be in a `## Preview` section.

**Comment by nstarman** [comment]
Would it be better if when `explicit-preview-rules` is on that even *existing* rules like SIM910 would need to be explicitly selected to turn on their preview mode?

**Comment by zanieb** [comment]
> Would it be better if when explicit-preview-rules is on that even existing rules like SIM910 would need to be explicitly selected to turn on their preview mode?

_Maybe_ but it adds a lot of complexity and it sounds just as confusing.

> preview option is for things still in development (otherwise it would just be released?), so it would be nice to have granular control over the previews even for existing rules.

Preview is for things that we want to get feedback on before we stabilize to reduce the number of changes that users in our stable track need to deal with. It's explicitly _not_ intended for things still in development and it is explicitly designed to be a relatively _global_ toggle to minimize the overhead of maintenance for us.
